For reference, here is the error I am seeing without this fix on g++ 11.4.0

```
-- The CXX compiler identification is GNU 11.4.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test HAVE_FLAG__ffile_prefix_map__home_djbn65_enchantum_build_cpp17_Release__deps_catch2_src__
-- Performing Test HAVE_FLAG__ffile_prefix_map__home_djbn65_enchantum_build_cpp17_Release__deps_catch2_src__ - Success
-- enchantum tests are being ran at runtime
-- Performing Test HAS_STD_FORMAT
-- Performing Test HAS_STD_FORMAT - Failed
-- This compiler does not support <format> header not running std_format.cpp tests
-- Configuring done
-- Generating done
-- Build files have been written to: /home/djbn65/enchantum/build-cpp17-Release
[126/128] Building CXX object tests/CMakeFiles/tests.dir/scoped.cpp.o
FAILED: tests/CMakeFiles/tests.dir/scoped.cpp.o 
/usr/bin/g++ -DCATCH_CONFIG_ENABLE_OPTIONAL_STRINGMAKER -DCATCH_CONFIG_RUNTIME_STATIC_REQUIRE -DENCHANTUM_ENABLE_MSVC_SPEEDUP=1 -I/home/djbn65/enchantum/tests/third_party -I/home/djbn65/enchantum/enchantum/include -I/home/djbn65/enchantum/build-cpp17-Release/_deps/catch2-src/src/catch2/.. -I/home/djbn65/enchantum/build-cpp17-Release/_deps/catch2-build/generated-includes -O3 -DNDEBUG -Werror -Wall -Wextra -Wshadow -Wconversion -Wpedantic -MD -MT tests/CMakeFiles/tests.dir/scoped.cpp.o -MF tests/CMakeFiles/tests.dir/scoped.cpp.o.d -o tests/CMakeFiles/tests.dir/scoped.cpp.o -c /home/djbn65/enchantum/tests/scoped.cpp
In file included from /usr/include/c++/11/string_view:41,
                 from /home/djbn65/enchantum/tests/case_insensitive.hpp:3,
                 from /home/djbn65/enchantum/tests/test_utility.hpp:2,
                 from /home/djbn65/enchantum/tests/scoped.cpp:1:
In static member function ‘static constexpr int std::char_traits<char>::compare(const char_type*, const char_type*, std::size_t)’,
    inlined from ‘constexpr int std::basic_string_view<_CharT, _Traits>::compare(std::basic_string_view<_CharT, _Traits>) const [with _CharT = char; _Traits = std::char_traits<char>]’ at /usr/include/c++/11/string_view:315:34,
    inlined from ‘constexpr bool std::operator==(std::basic_string_view<_CharT, _Traits>, std::basic_string_view<_CharT, _Traits>) [with _CharT = char; _Traits = std::char_traits<char>]’ at /usr/include/c++/11/string_view:537:53,
    inlined from ‘enchantum::scoped::details::remove_scope_or_empty(std::string_view, std::string_view)::<lambda(auto:6, auto:7)> [with auto:6 = std::basic_string_view<char>; auto:7 = std::basic_string_view<char>]’ at /home/djbn65/enchantum/enchantum/include/enchantum/scoped.hpp:24:82,
    inlined from ‘constexpr std::string_view enchantum::scoped::details::remove_scope_or_empty(std::string_view, std::string_view)’ at /home/djbn65/enchantum/enchantum/include/enchantum/scoped.hpp:25:23,
    inlined from ‘constexpr std::optional<_Tp> enchantum::scoped::details::scoped_cast_functor<E, <anonymous> >::operator()(std::string_view) const [with E = CStyleFromA_To_G; typename std::enable_if<is_enum_v<E>, int>::type <anonymous> = 0]’ at /home/djbn65/enchantum/enchantum/include/enchantum/scoped.hpp:54:54,
    inlined from ‘void CATCH2_INTERNAL_TEMPLATE_TEST_0() [with TestType = CStyleFromA_To_G]’ at /home/djbn65/enchantum/tests/scoped.cpp:16:5:
/usr/include/c++/11/bits/char_traits.h:389:32: error: ‘int __builtin_memcmp_eq(const void*, const void*, long unsigned int)’ specified bound 16 exceeds source size 14 [-Werror=stringop-overread]
  389 |         return __builtin_memcmp(__s1, __s2, __n);
      |                ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
In file included from /home/djbn65/enchantum/enchantum/include/enchantum/enchantum.hpp:7,
                 from /home/djbn65/enchantum/tests/scoped.cpp:4:
/home/djbn65/enchantum/enchantum/include/enchantum/entries.hpp: In function ‘void CATCH2_INTERNAL_TEMPLATE_TEST_0() [with TestType = CStyleFromA_To_G]’:
/home/djbn65/enchantum/enchantum/include/enchantum/entries.hpp:210:25: note: source object declared here
  210 |   inline constexpr auto reflection_data_string_storage = details::reflection_data_impl<E, NullTerminated>.strings;
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
[127/128] Building CXX object tests/CMakeFiles/tests.dir/enchantum.cpp.o
ninja: build stopped: subcommand failed.
```

--- a/enchantum/include/enchantum/scoped.hpp
+++ b/enchantum/include/enchantum/scoped.hpp
@@ -21,7 +21,11 @@ namespace scoped {
 
     constexpr string_view remove_scope_or_empty(string_view string, const string_view type_name) noexcept
     {
-      const auto starts_with = [](auto a, auto b) { return a.substr(0, b.size()) == b; };
+      const auto starts_with = [](auto a, auto b) {
+        // The a.size() >= b.size() check is not necessary here, but it is needed
+        // to silence a warning in gcc 11.
+        return a.size() >= b.size() && a.substr(0, b.size()) == b;
+      };
       if (!starts_with(string, type_name))
         return string_view();
       string.remove_prefix(type_name.size());

